server.js

```diff
@@ -1,23 +1,110 @@
 const express = require("express");
 const cors = require("cors");
+const bodyParser = require("body-parser");
 
 const app = express();
 
-app.use(cors());
+app.use(cors({
+        origin: 'http://localhost:3000',
+    }
+));
+
+
+app.use(bodyParser.urlencoded({ extended: true }));
 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
+  // timeSent:"18/05/1985"
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+let messages = [welcomeMessage]
+
+
+// level3 [5] read only text whose text contains a given substring
+// teniolao-cyf-chat-server.glitGlitchch.me/messages/search?text=express
+app.get("/messages/search", (req, res)=> {
+  const { term } = req.query
+  console.log(term)
+  
+  const filterMessages = messages.filter(message => message.text.toLowerCase().includes(term.toLowerCase()))
+  console.log(filterMessages)
+  res.send(filterMessages);
+});
+
+//[2] read all messages
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+//  Read only the most recent 10 messages:
+
+app.get("/messages/latest",(req,res)=>{
+  res.send(messages.slice(-10))
+})
+
+//[3] read one message specified by id
+app.get("/messages/:id", function (req, res) {
+  const id = req.params.id;
+  messages = messages.filter(
+    (message) => message.id === Number(id)
+  );
+    res.status(200).json(messages);
+});
+
+app.get("/messages/search",(req,res)=>{
+  let search=req.query.term
```

Comment:
These values don't change, so you can declare them as const instead.

The answers on this SO question give a good overview of why this is a good thing to do : https://stackoverflow.com/questions/136880/sell-me-on-const-correctness